### PR TITLE
refactor(Core): apply clang-tidy modernize-use-equals-default

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
- Part of https://github.com/azerothcore/azerothcore-wotlk/issues/3820
- Apply `clang-tidy` with `modernize-use-equals-default`
https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-equals-default.html

### How to test the changes

This PR is only about refactoring, it is then not meant to change anything of the program behaviour.
To test this, it is sufficient to:

- (Not mandatory but would be nice to build with `-DTOOLS=1` [extract the client files](https://www.azerothcore.org/wiki/Extract-Client-Data) and use the newly extracted data to run the server)
- Run the server
- Access the game and play around a bit
- All good so far? Then this should be good to go

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR- Part of https://github.com/azerothcore/azerothcore-wotlk/issues/3820
- Apply `clang-tidy` with `modernize-use-nullptr`
https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-nullptr.html

### How to test the changes

This PR is only about refactoring, it is then not meant to change anything of the program behaviour.
To test this, it is sufficient to:

- (Not mandatory but would be nice to build with `-DTOOLS=1` [extract the client files](https://www.azerothcore.org/wiki/Extract-Client-Data) and use the newly extracted data to run the server)
- Run the server
- Access the game
- All good so far? Then this should be good to go

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by JamesGent** [comment]
Approved.

Desktop 02-12-2020 14-17-39-750

**Comment by FrancescoBorzi** [comment]
@JamesGent thank you for testing this! Waiting for a code review to merge this.

File: src/common/Threading/LockedQueue.h [hunk 1]
```diff
@@ -37,8 +37,7 @@ namespace ACE_Based
 
         //! Destroy a LockedQueue.
         virtual ~LockedQueue()
```
Comment (Winfidonarleyan) [inline src/common/Threading/LockedQueue.h:39]: maybe add `= default;` here? and other

**Review by Winfidonarleyan — APPROVED** [review]

File: src/common/Threading/LockedQueue.h (hunk 1 above)
Comment (FrancescoBorzi) [reply src/common/Threading/LockedQueue.h]: @Winfidonarleyan right... done, thanks!

**Pull request merged**